File: LeetCode/Easy/0566. Reshape the Matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        int m = mat.size();
        int n = mat[0].size();

        if(m*n != r*c) return mat; // not legal

        vector<vector<int>> newMatrix;
        int curX = 0; int curY = 0;
        for(int i = 0; i < r; i++) {
            vector<int> newRow;
            for(int j = 0; j < c; j++) {
                newRow.push_back(mat[curX][curY]);
                curY++;
                if(curY == n) {
                    curX++;
                    curY = 0;
                }
            }
            newMatrix.push_back(newRow);
        }
        return newMatrix;
    }
};

File: LeetCode/Easy/0566. Reshape the Matrix.py
class Solution(object):
    def matrixReshape(self, mat, r, c):
        """
        :type mat: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        old_r, old_c = len(mat), len(mat[0])
        if old_r * old_c != r * c: return mat

        ans = [[-1]*c for _ in range(r)]

        new_i, new_j = 0,0
        for i in range(old_r):
            for j in range(old_c):
                ans[new_i][new_j] = mat[i][j]
                new_j += 1

                if new_j == c:
                    new_i += 1
                    new_j = 0
        return ans

        
